utils.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const isObject = (item) => {
+  return (item && typeof item === 'object' && !Array.isArray(item) && item !== null);
+};
+
+const mergeDeep = (target, source) => {
```

Comment:
Does this do an full deep merge? Or is it more of a copy objects? It also has a bit of a pyramid-of-doom to it, so it might be nice to break it out into a `mergeObject` function for the code in the `if`.